lib/services/transaction.js

```diff
@@ -154,10 +155,36 @@ TransactionService.prototype._confirmInput = function(ops, block, transaction) {
           }
         });
       }
-    }
+    });
   };
 };
 
+TransactionService.prototype._getAddressForInput = function(input) {
+  var script = input.script;
+  var self = this;
+
+  if (script.isPublicKeyHashIn()) {
+    var hash = bitcore.crypto.Hash.sha256ripemd160(script.chunks[0].buf);
+    return Promise.resolve(new bitcore.Address(
+      hash, bitcore.Networks.defaultNetwork, bitcore.Address.PayToPublicKeyHash
+    ));
+  } else if (script.isPublicKeyIn()) {
+    return self.getTransaction(input.prevTxId.toString('hex')).then(function(transaction) {
+      var outputScript = transaction.outputs[input.outputIndex].script;
+      if (outputScript.isPublicKeyOut()) {
+        return Promise.resolve(new bitcore.Address(
+          bitcore.crypto.Hash.sha256ripemd160(outputScript.chunks[0].buf),
+          bitcore.Networks.defaultNetwork, bitcore.Address.PayToPublicKeyHash
```

Comment:
I see the use of `bitcore.Networks.defaultNetwork` in many parts of the app. Are we going to change that variable to allow `testnet`?